header/p2p/session.go

```diff
@@ -130,14 +131,13 @@ func (s *session) doRequest(
 	req *p2p_pb.ExtendedHeaderRequest,
 	headers chan []*header.ExtendedHeader,
 ) {
+	ctx, cancel := context.WithTimeout(ctx, time.Second*3) // todo: extract into param
+	defer cancel()
+
 	r, size, duration, err := sendMessage(ctx, s.host, stat.peerID, s.protocolID, req)
 	if err != nil {
-		if err == context.Canceled || err == context.DeadlineExceeded {
-			return
-		}
 		log.Errorw("requesting headers from peer failed.", "failed peer", stat.peerID, "err", err)
 		select {
-		case <-ctx.Done():
```

Comment:
why? we will request headers from another peer. Previous one, will not be added

Reply:
because then request won't be retried!

Reply:
ctx.Done succeds and the `case s.reqCh <- req` does not do anything

Reply:
yes, correctly. Sorry, I was blind on Friday